docs/HyperIndex/Advanced/loaders.md

```diff
@@ -0,0 +1,100 @@
+---
+id: loaders
+title: Loaders
+sidebar_label: Loaders
+slug: /loaders
+---
+
+## Overview
+
+Loaders provide an efficient way to define and load entities from the database, optimizing performance by minimizing I/O operations. They also enable you to load arrays of entities based on specific field values.
+
+Loaders are designed to reduce I/O, which is often the primary performance bottleneck in an indexer. By using loaders, you can group multiple requests into a single database round trip before processing a batch of events. The retrieved values are cached in memory, allowing the batch to be processed entirely in memory.
+
+## Example: Refactoring a Simple Handler
+
+Let's explore how to convert a basic handler to use a loader. Below is an example of a handler that loads values at runtime, on demand. Note: Built-in caching still ensures that reloading the same value multiple times in a batch doesn't result in multiple database hits.
+
+```ts
+ERC20.Transfer.handler(async ({ event, context }) => {
+  const sender = await context.Account.get(event.params.from);
+  const receiver = await context.Account.get(event.params.to);
+  // ...Logic to update sender and receiver accounts
+});
+```
+
+### Refactoring with a Loader
+
+Now, let's refactor this handler to use a loader:
+
+```ts
+ERC20.Transfer.handlerWithLoader({
+  loader: async ({ event, context }) => {
+    const sender = await context.Account.get(event.params.from);
+    const receiver = await context.Account.get(event.params.to);
```

Comment:
I'd probably suggest using `Promise.all`, but there's a chance that it'll increase complexity of the example

Reply:
Ah, I see that you do this below :)